paasta_tools/cli/cmds/status.py

```diff
@@ -1236,13 +1256,167 @@ def get_version_table_entry(
     humanized_start_time = humanize.naturaltime(start_datetime)
     entry = [f"{version_name} - Started {start_datetime} ({humanized_start_time})"]
     replica_states = get_replica_states(replicaset.pods)
-    entry.append(f"  Replica States: {replica_states}")
+    replica_states = sorted(replica_states, key=lambda s: s[1].create_timestamp)
+
+    if replica_states:
+        # If no replica_states, there were no pods found
+        replica_state_list = []
+        for state in ReplicaState:
+            this_state_replicas = [rs for rs in replica_states if rs[0] == state]
+            if this_state_replicas:
+                replica_state_list.append(
+                    state.color(f"{len(this_state_replicas)} {state.value}")
+                )
+
+        entry.append(f"  Replica States: {' / '.join(replica_state_list)}")
+        if not verbose:
+            unhealthy_replicas = [rs for rs in replica_states if rs[0].is_unhealthy()]
+            if unhealthy_replicas:
+                entry.append(f"    Unhealthy Replicas:")
+                replica_table = create_replica_table(unhealthy_replicas)
+                for line in replica_table:
+                    entry.append(f"      {line}")
+        else:
+            replica_table = create_replica_table(replica_states)
+            for line in replica_table:
+                entry.append(f"    {line}")
     return entry
```

Comment:
nitpicky suggestion that I think might make this easier to deal with (feel free to take it or leave it): instead of having `get_replica_states` return tuples of `(state, pod)`, make it return a map of `pod` to `state`.  Then you can get the count of pods in each state with `Counter(pod_to_state.values())` (`Counter` is part of the `collections` module).  That lets you avoid accessing the tuples by index and also from having to go through the pods once per state.

Reply:
I wasn't able to convert to a dict where keys were pods since pods are unhashable types, but I did use `Counter` to generate the summary line with counts per state, and where we are working with the tuples I used named variables instead of accessing by index where I could (except for sorting the list, unfortunately).